src/fields.jl

```diff
@@ -52,3 +52,215 @@ Field(LX, data, grid::Grid1D) = Field1D(LX, data, grid)
 Constructs a 2D field of `data` at location `(LX, LY)` on `grid`.
 """
 Field(LX, LY, data, grid::Grid2D) = Field2D(LX, LY, data, grid)
+
+############################################################################
+#-------------------------------Interpolation------------------------------#
+
+𝐼xᶠ(i::Int, f::Field1D{Face})   = (f.data[i] + f.data[i+1]) / 2
+𝐼xᶜ(i::Int, f::Field1D{Centre}) = (f.data[i] + f.data[i-1]) / 2
+
+𝐼xᶠ(i::Int, j::Int, f::Field2D{Face, Centre})   = (f.data[i, j] + f.data[i+1, j]) / 2
+𝐼xᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i-1, j]) / 2
+𝐼yᶠ(i::Int, j::Int, f::Field2D{Centre, Face})   = (f.data[i, j+1] + f.data[i, j]) / 2
+𝐼yᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i, j-1]) / 2
```

Comment:
```suggestion
𝐼xᶠ(i, f::Field1D{Face})   = (f.data[i] + f.data[i+1]) / 2
𝐼xᶜ(i, f::Field1D{Centre}) = (f.data[i] + f.data[i-1]) / 2

𝐼xᶠ(i, j, f::Field2D{Face, Centre})   = (f.data[i, j] + f.data[i+1, j]) / 2
𝐼xᶜ(i, j, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i-1, j]) / 2
𝐼yᶠ(i, j, f::Field2D{Centre, Face})   = (f.data[i, j+1] + f.data[i, j]) / 2
𝐼yᶜ(i, j, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i, j-1]) / 2
```

Reply:
Typing this args does not provide anything other than get a somehow useful error in case non-integer values are accidentally passed into these functions. I remember a discussion in the Julia docs about this but I can't seem to find it now...

Reply:
@dhruvbhagtani, what
```julia
𝐼xᶠ(i, f::Field1D{Face})   = (f.data[i] + f.data[i+1]) / 2
```
is supposed to do. What does the superscript `f` denote?

When I initially wrote this in #7 I had written,
```Julia
𝐼ᶠ(i, f::Field1D{Center}) = (f.data[i] + f.data[i-1]) / 2
𝐼ᶜ(i, f::Field1D{Face})   = (f.data[i+1] + f.data[i]) / 2
```
which implied that that `𝐼ᶠ` stands for "interpolate on faces" and thus it takes `Field1D{Center}` as arguments. You seem to have changed the way this works, which is OK. I'm wondering though which of the two versions is more informative/useful?

Also, definitely for 2D fields, two superscripts will be needed. The way you have written them now I can't intuitively sniff out what they mean. If we were to follow your convention then it should be:
```Julia
𝐼xᶠᶜ(i, j, f::Field2D{Face, Centre})   = (f.data[i, j] + f.data[i+1, j]) / 2
𝐼xᶜᶜ(i, j, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i-1, j]) / 2
𝐼yᶜᶠ(i, j, f::Field2D{Centre, Face})   = (f.data[i, j+1] + f.data[i, j]) / 2
𝐼yᶜᶜ(i, j, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i, j-1]) / 2
```

Reply:
I'm starting to think we don't even need to use `x` and `y` in the function names. The `f`/`c` superscripts suffice, no?

Reply:
There should also be 𝐼ᶠᶠ, no?

Reply:
Yes, there needs to be an 𝐼ᶠᶠ. But for now, since we don't have any variables on this grid, I left it. We can add it later. The more urgent issue is to add an interpolation from U to V grid.

Reply:
Vorticity lives on `Face, Face`